docs/docs/guides/integrations/mcp.md

```diff
@@ -0,0 +1,236 @@
+
+- **MCP Hosts**: Programs like Claude Desktop, IDEs, or AI tools that want to access data through MCP.
+- **MCP Clients**: Protocol clients that maintain 1:1 connections with servers.
+- **MCP Servers**: Lightweight programs that expose specific capabilities through the standardized protocol.
+- **Data Sources**: Your computer's files, databases, services, or remote APIs that MCP servers can access.
+
+MCP servers provide three main types of capabilities:
+
+- **Tools**: Functions that can be called by clients
+- **Resources**: Static and dynamic data that can be read by clients
+- **Prompts**: Templated messages for consistent interactions
+
+## Why is tracing ability needed?
+
+As a developer you can fall into one of three categories:
+
+- **MCP server-side developer**: You want to expose multiple tools, resources, and prompts to the MCP client. You expose your existing application's tools, resources, etc., or you have built agents or have multiple agents orchestrated by an orchestrator agent. 
+
+- **MCP client-side developer**: You might want to plug your client-side application into multiple MCP servers. A core part of your client-side logic is making LLM calls to decide which tool to call or which resource to fetch.
+
+- **MCP server and client developer**: You are developing both the server and the client.
+
+If you fall into the first two categories, you want to know when each tool is called, what the execution flow looks like, the token count, and latency of different components in your server or client-side logic. 
+
+If you are developing both the server and client, the ability to see a unified trace timeline (we don't yet capture server-client interaction) can help you quickly iterate through both server and client-side logic. 
+
+With an observability layer you can:
+
+- Quickly iterate through your application
+- Audit the workflow or execution logic
+- Identify bottlenecks
+
+## How the Weave Integration Works
+
+We automatically patch the key methods of [`mcp.server.fastmcp.FastMCP`](https://github.com/modelcontextprotocol/python-sdk/blob/b4c7db6a50a5c88bae1db5c1f7fba44d16eebc6e/src/mcp/server/fastmcp/server.py#L109) and [`mcp.ClientSession`](https://github.com/modelcontextprotocol/python-sdk/blob/b4c7db6a50a5c88bae1db5c1f7fba44d16eebc6e/src/mcp/client/session.py#L84) class with a [`weave.op()`](../tracking/ops.md) decorator.
+
+We trace the following key components -- [**Tools**](https://modelcontextprotocol.io/docs/concepts/tools), [**Resources**](https://modelcontextprotocol.io/docs/concepts/resources), [**Prompts**](https://modelcontextprotocol.io/docs/concepts/prompts)
+
+## Installation
+
+To use the MCP integration with Weave, you'll need to install both weave and the MCP package:
+
+```bash
+pip install -qq mcp[cli] weave
+```
+
+## Quickstart Guide
+
+### Prerequisites
+
+Make sure you have both Weave and MCP installed as described in the Installation section.
+
+### Using the Example Code
```

Comment:
Make sure that readers don't think that this is the integration, that its just an example of the integration in action. maybe make that even clearer, my fear is that someone skimming the docs will see having to run .py scripts as part of the integration and bail